src/hera.cpp

```diff
@@ -50,6 +52,49 @@ struct hera_instance : evm_instance {
 
 namespace {
 
+#if HERA_EVM2WASM_JS
+// NOTE: assumes that pattern doesn't contain any formatting characters (e.g. %)
+string mktemp_string(string pattern) {
+  char tmp[PATH_MAX] = { 0 };
```

Comment:
You don't need this tmp on stack. Just resize string to `PATH_MAX`, then check the length of path with `strlen` and shrink the string back.

Reply:
Not sure I follow. `mktemp` modified the template in place. I guess I could:
```
string ret(pattern);
mktemp(ret.c_str());
return ret;
```

Reply:
You have to use `ret.data()`, not `ret.c_str()`. In C++11 the `.data()` will have the null terminator.

Reply:
Yeah, whichever has the null terminator.

Is that the preferred way then?

Reply:
The `.c_str()` returns const pointer, not useful here.

Reply:
> 
Accessing the value at data()+size() produces undefined behavior: There are no guarantees that a null character terminates the character sequence pointed by the value returned by this function. See string::c_str for a function that provides such guarantee.

Reply:
> Returns a pointer to an array that contains a null-terminated sequence of characters (i.e., a C-string) representing the current value of the string object.
>
>This array includes the same sequence of characters that make up the value of the string object plus an additional terminating null-character ('\0') at the end.
>
>The pointer returned points to the internal array currently used by the string object to store the characters that conform its value.
>
>Both string::data and string::c_str are synonyms and return the same value.

http://www.cplusplus.com/reference/string/string/data/

Reply:
Damn didn't see mine is valid for C++98 and yours is C++11. Ok, fine then, <11 was just bad.

Reply:
Cannot use `.data()` as that is `const` and `mktemp` requires non-const.